src/compiler/binder.ts

```diff
@@ -1154,6 +1154,9 @@ namespace ts {
                     else if (p.kind === SyntaxKind.ShorthandPropertyAssignment) {
                         bindAssignmentTargetFlow((<ShorthandPropertyAssignment>p).name);
                     }
+                    else if (p.kind === SyntaxKind.SpreadElementExpression) {

```

Comment:
I've been debating renaming this to remove the "Expression" part, since the node itself is not an Expression, and can only be used in an array literal, argument list, or binding pattern.